command-design.md

```diff
@@ -0,0 +1,224 @@
+
+```bash
+Builds the app. Example: corectl build --connections ./myconnections.yml --script ./myscript.qvs
+
+Usage:
+  corectl build [flags]
+
+Flags:
+  -a, --app string       App name including .qvf file ending.
+  -c, --config string    path/to/config.yml where parameters can be set instead of on the command line
+  --connections string   path/to/connections.yml that contains connections that are used in the reload. Note that when specifying connections in the config file they arspecified inline, not as a file reference!
+  -e, --engine string    URL to engine (default "localhost:9076")
+  -h, --help             help for build
+      --objects string   A list of object json paths
+      --script string    path/to/reload-script.qvs that contains a qlik reload script. If omitted the last specified reload script for the current app is reloaded
+      --silent           Do not log reload progress
+      --ttl string       Engine session time to live (default "30")
+
+Global Flags:
+  -v, --verbose         Logs extra information
+```
+
+### Eval Command
+
+```bash
+Evalutes a list of measures and dimensions. Meaures are separeted from dimensions by the "by" keyword. To omit dimensions and only use measures use "*" as dimension: eval <measures> by *
+
+Usage:
+  corectl eval <measure 1> [<measure 2...>] by <dimension 1> [<dimension 2...] [flags]
+
+Flags:
+  -a, --app string      App name including .qvf file ending. If no app is specified a session app is used instead.
+  -c, --config string   path/to/config.yml where parameters can be set instead of on the command line
+  -e, --engine string   URL to engine (default "localhost:9076")
+  -h, --help            help for eval
+      --ttl string      Engine session time to live (default "30")
+
+Global Flags:
+  -v, --verbose         Logs extra information
+```
+
+This is the same as current design, only flags have been moved from global to the command.
+
+### Get Command
+
+```bash
+Lists one or several resources
+
+Usage:
+  corectl get [command]
+
+Available Commands:
+  apps        Prints a list of all apps available in the current engine
+  assoc       Print table associations summary
+  data        Evalutes the hypercube data of an object defined by the --object parameter. Note that only basic hypercubes like straight tables are supported
+  field       Shows content of a field
+  fields      Print field list
+  keys        Print key-only field list
+  layout      Evalutes the hypercube layout of an object defined by the --object parameter
+  meta        Shows metadata about the app
+  objects     Prints a list of all objects in the current app
+  properties  Prints the properties of the object identified by the --object flag
+  script      Print the reload script
+  status      Prints status info about the connection to engine and current app
+  tables      Print tables summary
+
+Flags:
+  -c, --config string   path/to/config.yml where parameters can be set instead of on the command line
```

Comment:
This section will need to specify app as well? 
```suggestion
    -a, --app string      App name including .qvf file ending. If no app is specified a session app is used instead.
```

Reply:
No, not on the `get` level, since all commands there does not use the `app` parameter.

The `app` param would be one step `down` as in `kubectl get tables --app my-cool.qvf`

So the `app` param is a flag for the `get tables` command